src/main/kotlin/org/jitsi/jicofo/xmpp/XmppServices.kt

```diff
@@ -80,3 +99,62 @@ class XmppServices(xmppProviderFactory: XmppProviderFactory) {
 }
 
 enum class XmppConnectionEnum { Client, Service }
+
+class AvModerationHandler(val xmppConnection: AbstractXMPPConnection) : StanzaListener, RegistrationListener {
+    var conferenceStore: ConferenceStore = EmptyConferenceStore()
+    private val jsonParser = JSONParser()
+    private var avModerationAddress: String? = null
+
+    override fun processStanza(stanza: Stanza) {
+        if (!stanza.from.equals(avModerationAddress)) {
+            return
+        }
+
+        val jsonMessage = stanza.getExtension<JsonMessageExtension>(
+            JsonMessageExtension.ELEMENT_NAME, JsonMessageExtension.NAMESPACE
+        ) ?: return Unit.also {
+            logger.warn("Skip processing stanza without JsonMessageExtension")
+        }
+
+        val incomingJson = jsonParser.parse(jsonMessage.json) as JSONObject
+        if (incomingJson["type"] == "av_moderation") {
+            val conferenceJid = JidCreate.entityBareFrom(incomingJson["room"] as String)
+
+            val conference = conferenceStore.getConference(conferenceJid) ?: return Unit.also {
+                logger.warn("Not processing message for invalid conferenceJid=$conferenceJid")
+            }
+
+            val enabled = incomingJson["enabled"]
```

Comment:
Why do we encode a boolean as a string in json?

Reply:
Ah its coming from lua ... as we take it directly from the stanza, I will change that

Reply:
Done